core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -1866,12 +1830,27 @@ func (o *evmTxStore) ReapTxHistory(ctx context.Context, minBlockNumberToKeep int
 	var cancel context.CancelFunc
 	ctx, cancel = o.mergeContexts(ctx)
 	defer cancel()
-	qq := o.q.WithOpts(pg.WithParentCtx(ctx))
+
+	Batch := func(cb func(limit uint) (count uint, err error)) error {
+		offset := uint(0)
+		var limit uint = 1000
+		for {
+			count, err := cb(limit)
+			if err != nil {
+				return err
+			}
+			if count < limit {
+				return nil
+			}
+			offset += limit
+		}
+	}
```

Comment:
Why use a custom func instead of the helper? If the point is to shed the import, then we should introduce a new helper func in common.

Reply:
Fair enough! PR: https://github.com/smartcontractkit/chainlink-common/pull/411